src/common/SimplifyTransform.scala

```diff
@@ -127,11 +127,11 @@ trait SimplifyTransform extends internal.FatScheduling {
           case l: AbstractIfThenElse[_] => l
           case Reflect(l: AbstractIfThenElse[_], _, _) => l
         }
-        val cond2 = if (lhs != lhs2) mhs2.map (_.toIf.cond) reduceLeft { (s1,s2) => assert(s1==s2,"conditions don't agree: "+s1+","+s2); s1 }
+        val cond2 = if (lhs != lhs2) mhs2.map (infix_toIf(_).cond) reduceLeft { (s1,s2) => assert(s1==s2,"conditions don't agree: "+s1+","+s2); s1 }
```

Comment:
I would lean towards renaming these infix_ methods since they're not really infix anymore